components/hab/src/cli/hab/svc.rs

```diff
@@ -196,3 +205,139 @@ pub struct Load {
     #[serde(flatten)]
     shared_load: SharedLoad,
 }
+
+/// Update how the Supervisor manages an already-running
+/// service. Depending on the given changes, they may be able to
+/// be applied without restarting the service.
+#[derive(ConfigOpt, StructOpt, Deserialize)]
+#[configopt(attrs(serde))]
+#[serde(deny_unknown_fields)]
+#[structopt(name = "update", no_version, rename_all = "screamingsnake")]
+#[allow(dead_code)]
+pub struct Update {
+    #[structopt(flatten)]
+    #[serde(flatten)]
+    pkg_ident: PkgIdent,
+
+    #[structopt(flatten)]
+    #[serde(flatten)]
+    pub remote_sup: RemoteSup,
+
+    // This is some unfortunate duplication... everything below this
+    // should basically be identical to SharedLoad, except that we
+    // don't want to have default values, and everything should be
+    // optional.
```

Comment:
:thinking: On the one hand, I do like cleaning up the duplication. On the other hand, it would be nice to _not_ have the default value information in there (I believe that affects the generated CLI help, no?). Ensuring that the `Vec` field is made into an `Option` is particulary important here because it seems to be crucial for allowing users to remove binds, should that be desired.

If we can get that behavior, then I'm all for the change :+1: